pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -74,6 +72,10 @@ func updateAvailable(ac v1alpha1.Condition, prometheusConditions []monv1.Prometh
 		return ac
 	}
 
+	if prometheusAvailable.ObservedGeneration != prom.Generation {
+		return ac
+	}
+
```

Comment:
Should we leave a note for future devs on what the reasoning is for not updating the conditions?
I'd be :+1: on that since this PR already needed a bit of discussion about the motivation for this.
Otherwise this lgtm.

Reply:
Yeah why not. Let me add some explanation there.